src/sentry/tasks/sentry_apps.py

```diff
@@ -211,32 +224,38 @@ def _process_resource_change(action, sender, instance_id, retryer=None, *args, *
         org = Organization.objects.get_from_cache(
             id=Project.objects.get_from_cache(id=instance.project_id).organization_id
         )
-
-    installations = filter(
-        lambda i: event in i.sentry_app.events,
-        app_service.get_installed_for_organization(organization_id=org.id),
-    )
-
-    for installation in installations:
-        data = {}
-        if isinstance(instance, Event) or isinstance(instance, GroupEvent):
-            data[name] = _webhook_event_data(instance, instance.group_id, instance.project_id)
-        else:
-            data[name] = serialize(instance)
-
-        # Trigger a new task for each webhook
-        send_resource_change_webhook.delay(installation_id=installation.id, event=event, data=data)
+        assert org, "organization must exist to get related sentry app installations"
+        installations: list[RpcSentryAppInstallation] = [
+            installation
+            for installation in app_service.get_installed_for_organization(organization_id=org.id)
+            if event in installation.sentry_app.events
+        ]
+
+        for installation in installations:
+            data = {}
+            if isinstance(instance, (Event, GroupEvent)):
+                assert instance.group_id, "group id is required to create webhook event data"
+                data[name] = _webhook_event_data(instance, instance.group_id, instance.project_id)
+            else:
+                data[name] = serialize(instance)
+
+            # Trigger a new task for each webhook
+            send_resource_change_webhook.delay(
+                installation_id=installation.id, event=event, data=data
+            )
```

Comment:
There is a slight logical change here that this code will only be ran if the `instance` is a `Group | Event | GroupEvent` (i.e if the instance is an Activity we do nothing) but I believe this was the case with the previous iteration too (?) since we don't get an organization and therefore installation to work with.

Sidenote: Do we process Activity/Comments here ? I couldn't find any related code calling this func for those event types.

Reply:
> There is a slight logical change here that this code will only be ran if the instance is a Group | Event | GroupEvent (i.e if the instance is an Activity we do nothing) but I believe this was the case with the previous iteration too 

I don't think that difference in behavior will matter. Currently if Activity is used there will be an `AttributeError` for org.id which doesn't exist.

> Sidenote: Do we process Activity/Comments here ? I couldn't find any related code calling this func for those event types.

There are no tests covering Comments/Activity, and no call sites that I could find. Once we've finished moving the tasks we could add a metric/log message to validate that there are no indirect paths that use `Activity` and then remove this code if it is truly unused.